Lib/tarfile.py

```diff
@@ -357,14 +358,15 @@ def __init__(self, name, mode, comptype, fileobj, bufsize):
             fileobj = _StreamProxy(fileobj)
             comptype = fileobj.getcomptype()
 
-        self.name     = name or ""
-        self.mode     = mode
+        self.name = name or ""
+        self.mode = mode
         self.comptype = comptype
-        self.fileobj  = fileobj
-        self.bufsize  = bufsize
-        self.buf      = b""
-        self.pos      = 0
-        self.closed   = False
+        self.fileobj = fileobj
+        self.bufsize = bufsize
+        self.compresslevel = compresslevel
```

Comment:
Other attributes are used in multiple `_Stream` methods or are needed for simulating a file object. But `compresslevel` is used only in the constructor (`_init_write_gz()` is a factored out part of the constructor). It is not needed after creating a `_Stream` object.